### ISSUE TITLE
Generalize `useAssistant` and `AssistantResponse` to accept an `assistantId`

### ISSUE THREAD

**Issue by mfts** [vercel/ai#784]
### Feature Description

Add `assistantId` to `useAssistant` and `AssistantResponse`. 

### Use Case

I'm creating assistants on-demand and want to pass the assistantId into `useAssistant`, so it's sent to the the edge server along with `threadId` and `message`. 

Furthermore, I want to add the `assistantId` to `AssistantResponse` because assistantId is created dynamically. Current implementation relies on assistantId coming from .env

**Comment by tgonzales** [comment]
Hi @mfts Just like useChat you can use body to send customized data

**Comment by mfts** [comment]
@tgonzales I don't see how `useAssistant` takes a body argument? What am I missing?

https://github.com/vercel/ai/blob/8b4f7d1219d4a764a3a7226d546fb232d217dd93/packages/core/react/use-assistant.ts#L10-L13

**Comment by tgonzales** [comment]
You're right @mfts , I thought the implementation was closer to useChat, I believe the best way would be this, adding body and header to useAssistant would give more flexibility. I'll try to make a PR for this feature

**Comment by lgrammel** [comment]
@tgonzales I have some capacity to take this on. Let me now if you want to add it, otherwise I could take a look at it today or tomorrow.

**Comment by lgrammel** [comment]
Also, as a note, I would prefer not adding the `assistantId` to `useAssistant`, since this is an edge case and in most cases the assistant ID should be hidden on the server side. Instead I would implement the option to send arbitrary data with the `useAssistant` hook, so you can handle this special case (I can provide an example).

**Comment by holdenmatt** [comment]
Great to see a data prop is being added (thanks!). I'm building a prototype that uses multiple assistants, so having some way to specify an assistantId will be very useful!

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Feature**  
Confidence: **High**

### Rationale
This requests a new configuration capability for `useAssistant` so callers can send custom data such as an assistant ID. The discussion confirms it as an API enhancement rather than a current-behavior defect.

### Evidence
- The requester wants `useAssistant` to send a dynamically created `assistantId` with requests.
- A maintainer proposes adding support for arbitrary request data rather than a dedicated `assistantId`, noting this capability does not currently exist.
- Comments describe adding body/header or data options as increased flexibility, not as a failure of documented behavior.
